docs: rbac getting started guide (#48029)
gravitational/teleport: docs/pages/enroll-resources/kubernetes-access/kubernetes-access-examples.mdx

---
title: Examples of Role-Based Access Control in Kubernetes
description: RBAC examples with Kubernetes.
---

In Teleport's flexible RBAC system, you can define custom roles and granularly control what users can do within your Kubernetes clusters. 
Limit access to specific namespaces, enforce read-only permissions, or restrict administrative actions as needed.

> :Note:
> Please reference [Setting Up Teleport Access Controls for Kubernetes](../../enroll-resources/kubernetes-access/manage-access.mdx) before proceeding, if you have not done so already.

Reviewer: ```suggestion
<Notice type="tip">
Reference [Setting Up Teleport Access Controls for Kubernetes](../../enroll-resources/kubernetes-access/manage-access.mdx) before proceeding, if you have not done so already.
</Notice>
```
- Using our docs engine syntax to create a notice
- Removing "Please", which I don't think we need in technical writing

Here are a few common use cases. These are sample examples to get you started; for more in-depth coverage of roles in Kubernetes access, please visit the 
[Access Controls guide](./controls.mdx) for Kubernetes.

## Use case 1: Example role for developers

In many organizations, developers need limited access to Kubernetes clusters, typically restricted to their development environments. 

Below is an example of a role that provides developers with access to only the dev namespace:

> ```code

Reviewer: ```suggestion
```yaml
```

Reviewer: We should use `yaml` for other YAML examples as well so syntax highlighting works.

---
title: Examples of Role-Based Access Control in Kubernetes
description: RBAC examples with Kubernetes.
---

In Teleport's flexible RBAC system, you can define custom roles and granularly control what users can do within your Kubernetes clusters. 
Limit access to specific namespaces, enforce read-only permissions, or restrict administrative actions as needed.

<Admonition type="tip">

> Reference [Setting Up Teleport Access Controls for Kubernetes](../../enroll-resources/kubernetes-access/manage-access.mdx) before proceeding, if you have not done so already.

Reviewer 2: ```suggestion
Reference [Setting Up Teleport Access Controls for Kubernetes](./manage-access.mdx) before proceeding, if you have not done so already.
```

Wouldn't this be identical?

but does not allow any write operations:

```code
kind: role
version: v6
metadata:
  name: auditor-k8s-read-only
spec:
  allow:
    kubernetes_groups: ["auditor-group"]
    kubernetes_labels:
      "*": "*"  # Access to all clusters
    kubernetes_resources:
      - kind: pod
        verbs: ["get", "list", "watch"]
      - kind: service
        verbs: ["get", "list", "watch"]
  deny:
    kubernetes_resources:
      - kind: pod
        verbs: ["create", "update", "delete"]  # Deny any modification operations
      - kind: service
        verbs: ["create", "update", "delete"]
```

In this example, the auditor-group is allowed to access all namespaces (*), but only with `get`, `list`, and `watch` permissions (read-only). Modifications such as creating, updating, or deleting resources are denied.

## Use case 3: Admin access to all Kubernetes clusters

In organizations with DevOps or platform engineering teams, certain users need full administrative access to all Kubernetes clusters for management purposes. This role grants full access to all Kubernetes resources across all clusters, allowing the user to perform any action on any resource.

```code
kind: role
version: v6
metadata:

>   name: admin-k8s-access

Reviewer 3: might want to update this to role v7 now https://goteleport.com/docs/enroll-resources/kubernetes-access/manage-access/#define-a-role
